### PR TITLE
Add Batch PyRad2CSV

### PR DESCRIPTION
**Pull request by Tamookk**
This PR adds batch processing of exporting Pyradiomics data to a CSV file:

- This PR adds another tab to the batch processing window which allows the user to select the output CSV file path for PyRad2CSV.
- The functionality performs Pyrad2CSV on all patients in the selected folder, and saved the radiomics data into a single CSV file.
- The batch processing summary screen now includes a summary on PyRad2CSV functionality.
- The Github test has been updated to install plastimatch so that Pyrad2CSV can be properly tested.

### CODE REVIEW

**Comment by sjswerdloff** [comment]
Is this only intended for *nix environment s?
The commands invoked are based on strings that appear to be  *nix (e.g 'rm')

It's OK if this is expected to be os constrained.
Otherwise the python platform agnostic commands (potentially imported from a package) should be used.

**Comment by Tamookk** [comment]
This is not intended to be OS constrained, and it looks as though if you're not running a *nix OS then the delete command (the only one that would differ between platforms) is replaced with a Windows-specific one. I've been testing this code on Windows and everything seems to work okay, with Pyradiomics running properly and temporary files and folders being removed.

Using Python commands (e.g. `os.remove`) is definitely better for long-term maintainability, but if it ain't broke... 

The code that runs these commands in this PR was copied from the existing Pyradiomics code, also. We didn't want to change too much here in case something broke, somewhere.

File: src/Controller/BatchProcessingController.py [hunk 1]
```diff
@@ -320,6 +325,38 @@ def batch_dvh2csv_handler(self, interrupt_flag,
...
+        Handles creating, starting, and processing the results of batch
+        Pyrad2CSV.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param patient: The patient to perform this process on.
+        """
+        # Get current files
+        cur_patient_files = self.get_patient_files(patient)
```
Comment (eddieGong2604) [inline src/Controller/BatchProcessingController.py:340]: Same problem as previous PR here. Static method should not be called by an instance

File: src/Model/batchprocessing/BatchProcessPyRad2CSV.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+            self.summary = "INTERRUPT"
+            return False
+
+        if not self.ready:
+            self.summary = "SKIP"
+            return False
+
+        rtss_path = self.patient_dict_container.filepaths.get('rtss')
+        patient_id = self.patient_dict_container.dataset.get('rtss').PatientID
+        patient_id = self.clean_patient_id(patient_id)
```
Comment (eddieGong2604) [inline src/Model/batchprocessing/BatchProcessPyRad2CSV.py:67]: clean_patient_id is static method, should not be called by an instance

**Review by eddieGong2604 — COMMENTED** [review]
Please fix the problem of static method getting called by instance

File: src/Controller/BatchProcessingController.py (hunk 1 above)
Comment (Tamookk) [reply src/Controller/BatchProcessingController.py]: These have been fixed. Thank you for picking these up! Must have missed them before I created the PR.

File: src/Controller/BatchProcessingController.py [hunk 3]
```diff
@@ -320,6 +325,39 @@ def batch_dvh2csv_handler(self, interrupt_flag,
...
+        process = BatchProcessPyRad2CSV(progress_callback,
+                                        interrupt_flag,
+                                        cur_patient_files,
+                                        self.pyrad_output_path)
+        process.set_filename('PyRadiomics_' + self.timestamp + '.csv')
+        success = process.start()
+
+        # Set summary message
+        if success:
+            reason = "SUCCESS"
```
Comment (eddieGong2604) [inline src/Controller/BatchProcessingController.py:351]: If you can make summary messages enum or constants, and import them when you want to use that would be cleaner.

File: src/Model/batchprocessing/BatchProcessPyRad2CSV.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+    def start(self):
+        """
+        Goes through the steps of the PyRadCSV conversion.
+        """
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
```
Comment (eddieGong2604) [inline src/Model/batchprocessing/BatchProcessPyRad2CSV.py:60]: If you can make summary messages enum or constants, and import them when you want to use that would be cleaner.

**Review by JThomas2020 — APPROVED** [review]
Team 19, sorry for the delay, great job guys! everything works and code is high quality

File: src/Model/batchprocessing/BatchProcessPyRad2CSV.py [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+            patient_path, patient_id, patient_nrrd_file_path, mask_folder_path)
+
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        if radiomics_df is None:
+            self.summary = "PYRAD_NO_DF"
```
Comment (eddieGong2604) [inline src/Model/batchprocessing/BatchProcessPyRad2CSV.py:128]: If you can make summary messages enum or constants, and import them when you want to use that would be cleaner.

**Review by eddieGong2604 — COMMENTED** [review]
Team 23: Please consider making possible summary values constants instead of hardcode.
I'm still waiting for the pyrad2CSV to finish on my machine. How long does it normally take?

**Comment by Tamookk** [comment]
I'm not sure how well making summary values constants would work for some later PRs we are making, as some specific information is directly contained in these summary strings (such as numbers that need to be extracted). This information can't be passed over if the summary values are made constant. I will definitely look into this however, but not for this PR.

The amount of time Pyrad2CSV takes depends on how many datasets you perform it on, and how many ROIs are in these datasets. The more datasets and ROIs, the longer it takes. That is not due to how we coded this PR as the actual radiomics calculations are done by external program. My suggestion for testing would be to perform it on one dataset with a small amount of ROIs. The longest I have seen it take personally is about half an hour for 5 datasets with many large ROIs in them.

**Review by eddieGong2604 — APPROVED** [review]
Team 23 approves this PR

**Pull request merged**